godot-core/src/private.rs

```diff
@@ -239,18 +226,45 @@ pub fn extract_panic_message(err: Box<dyn std::any::Any + Send>) -> String {
     }
 }
 
-fn format_panic_message(msg: String) -> String {
+fn format_panic_message(_location: Option<&std::panic::Location<'_>>, mut msg: String) -> String {
+    if let Some(context) = get_gdext_panic_context() {
+        msg = format!("{msg}\nContext: {context}");
+    }
+
+    let prefix = if let Some(location) = _location {
+        format!("panic {}:{}", location.file(), location.line())
+    } else {
+        "panic".to_string()
+    };
+
     // If the message contains newlines, print all of the lines after a line break, and indent them.
     let lbegin = "\n  ";
     let indented = msg.replace('\n', lbegin);
 
     if indented.len() != msg.len() {
-        format!("[panic]{lbegin}{indented}")
+        format!("[{prefix}]{lbegin}{indented}")
     } else {
-        format!("[panic]  {msg}")
+        format!("[{prefix}]  {msg}")
     }
 }
 
+pub fn set_gdext_hook(godot_print: impl 'static + Send + Sync + Fn() -> bool) {
```

Comment:
I'd mention the most important trait first (meaning that it's a function), and lifetime at the end:
```suggestion
pub fn set_gdext_hook(godot_print: impl Fn() -> bool + Send + Sync + 'static) {
```
Or even a `where` clause with `F` parameter...